packages/core/src/events/input/load/vectorstore/chroma.ts

```diff
@@ -0,0 +1,333 @@
+import {
+  ChromaClient,
+  Collection as ChromaCollection,
+  CollectionMetadata as ChromaCollectionMetadata,
+  Metadata as ChromaMetadata,
+  IncludeEnum,
+  QueryResponse,
+  Where,
+} from 'chromadb';
+import { v4 as uuid } from 'uuid';
+
+import { maximalMarginalRelevance } from '../../../../utils/math.js';
+import { Context } from '../docs/context.js';
+import { BaseVectorStore, BaseVectorStoreField } from './base.js';
+
+export interface ChromaVectorStoreField extends BaseVectorStoreField {
+  url: string;
+
+  numDimensions?: number;
+
+  collectionName?: string;
+
+  collectionMetadata?: ChromaCollectionMetadata;
+
+  filter?: ChromaVectorStoreFilterType;
+}
+
+export interface ChromaDeleteParams<T> {
+  ids?: string[];
+  filter?: T;
+}
+
+export type ChromaVectorStoreFilterType = {
+  where?: Where;
+  includeEmbeddings?: boolean;
+};
+
+export class ChromaVectorStore
+  extends BaseVectorStore
+  implements ChromaVectorStoreField
+{
+  declare FilterType: ChromaVectorStoreFilterType;
+
+  _isSerializable = true;
+
+  url: string;
+
+  numDimensions?: number | undefined;
+
+  collectionName?: string | undefined;
+
+  collectionMetadata?: ChromaCollectionMetadata | undefined;
+
+  filter?: ChromaVectorStoreFilterType | undefined;
+
+  private _client: ChromaClient;
+
+  _vectorstoreType(): string {
+    return 'chroma';
+  }
+
+  constructor(fields?: Partial<ChromaVectorStoreField>) {
+    super(fields);
+
+    this.url = fields?.url ?? 'http://localhost:8000';
+    this.numDimensions = fields?.numDimensions;
+    this.collectionName = fields?.collectionName;
+    this.collectionMetadata = fields?.collectionMetadata;
+    this.filter = fields?.filter;
+  }
+
+  async addVectors(
+    embeddings: number[][],
+    context: Context[],
+    options?: { ids?: string[] }
+  ): Promise<void> {
+    if (embeddings.length === 0) {
+      return;
+    }
+
+    if (embeddings.length !== context.length) {
```

Comment:
I am pretty sure we have a method for this, but just want to double check, when we go through splitting, since now our input is a string, (we seem to have eliminated) the Document class usage, so when we go through splitting, do we pass in context.length, to make sure that each chunk of the splitted text will have a length <= context.length?

Reply:
where will the splitting occurs then, since now we take in the string as input rather than Document object which I remember was used after calling load_and_split from the langchain library?

Reply:
so, we expect our users to call them by themselves, right?

Reply:
yes